src/launchpad/server.py

```diff
@@ -107,15 +109,35 @@ async def create_app(self) -> Application:
         return app
 
     async def health_check(self, request: Request) -> Response:
-        """Basic health check endpoint."""
-        return web.json_response(
-            {
-                "status": "ok",
-                "service": "launchpad",
-                "version": "0.0.1",
-                "environment": self.config["environment"],
-            }
-        )
+        """Health check endpoint that checks all service components."""
+        try:
+            # Get health status from the service if callback is provided
+            if self.health_check_callback:
+                health_data = await self.health_check_callback()
+
+                # Determine HTTP status code based on health status
+                if health_data.get("status") == "ok":
```

Comment:
Does this need more resilient handling or will it always handle JSON data? I believe this is typed as `Any` and then we are assuming it returns JSON.